src/hooks/use-settings-data.ts

```
@@ -0,0 +1,49 @@
+
+interface UseSettingsDataOptions {
+    section: string;
+    defaultTab: string;
+    dataLoaders: {
+        [key: string]: () => Promise<any>;
+    };
+}
+
+export function useSettingsData({ section, defaultTab, dataLoaders }: UseSettingsDataOptions) {
+    const { activeTab, handleTabChange } = useSettingsTab({ section, defaultTab });
+    const [data, setData] = useState<{ [key: string]: any }>({});
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)